components/bio-formats-plugins/src/loci/plugins/out/Exporter.java

```diff
@@ -656,8 +656,14 @@ else if (proc instanceof ShortProcessor) {
             int start = doStack ? 0 : imp.getCurrentSlice() - 1;
             int end = doStack ? size : start + 1;
 
-            boolean littleEndian =
-                    !w.getMetadataRetrieve().getPixelsBinDataBigEndian(0, 0).booleanValue();
+            boolean littleEndian = false;
+            if (w.getMetadataRetrieve().getPixelsBigEndian(0) != null)
+            {
+              littleEndian = !w.getMetadataRetrieve().getPixelsBigEndian(0).booleanValue();
+            }
+            else if (w.getMetadataRetrieve().getPixelsBinDataCount(0) == 0) {
+              littleEndian = !w.getMetadataRetrieve().getPixelsBinDataBigEndian(0, 0).booleanValue();
+            }
```

Comment:
This two step pattern is so common, I do wonder if we shouldn't implement it directly inside `getPixelsBigEndian` so it works correctly all the time for every caller?